docs/filestructure.rst

```diff
@@ -0,0 +1,228 @@
+.. _filestructure:
+
+==================================
+File structure example & first app
+==================================
+
+We will deal here with a suggestion how you could structure your files and folders for your individual atk4 project.
+Let's assume you are planning to create at least several pages with some models and views. This example can be expanded
+and modified to your needs and shows just one concept of how to setup an atk4 project.
+
+
+File structure example
+======================
+
+This file structure is a recommendation and no must. It is a best practice example.
+Feel free to experiment with it and find the ideal file structure for your project.
+
+* config
+
+    * db.php
+
+* public_html
+
+    * vendor (contains all needed composer modules - don't touch them)
```

Comment:
vendor must be in root like config and never in public folder

Reply:
Reason why it's in public_html is because I'm loading JS and CSS frameworks as well. These have to be accessible through the domain. Therefore vendor has to be in public folder in my case. If you just use atk4, you could move vendor outside of the public_html folder that's true. But I've tried to bring in a real world example...as the other examples mostly have a limited scope and are good for small projects with just a few repositories. 
But as your app grows, you need a more flexible structure imho.

Sure, you could have 2 Composer files, one in public and one hidden from public. But if a repository depends e.g. on PHPSpreadsheet it would be downloaded twice, one time in public vendor and one time in hidden vendor.
This is imho unnecessary...but I'm open for discussion of better practice ;)

Reply:
IMHO JS and CSS can be copied in *public/assets* folder in place of exposing all the vendor directory, but leaving apart the actual problem of needing of rebuild css and js, i think is better to leave the default assets that are defined in $app->cdn which are loaded by default

Reply:
Default assets means there's no flexibility. What if you have a client that wants flexibility?
Just copying Composer assets to other directories clutters up everything. You can't maintain them properly and they can't be autoloaded. 
As an example: I'd like to use "select2" which replaces conventional select boxes.

Reply:
IMHO if you want to use select2, YOU need to tweak the standard configuration and there is no point on adding complexity in a getting started project.

- app
- config
- - db.php
- models
- public
- - index.php
- vendor
- init.php
- composer.json

if you start to put some customization, can be disorienting


- app
- - User
- - - Grid.php
- - - Form.php
- - Profile.php
- - Login.php
- - Logout.php
- config
- - db.php
- models
- - User.php
- public
- - index.php
- - admin
- - - index.php
- vendor
- init.php
- composer.json

With admin can be, but already here there are many ways of defining a structure, for that i want to discuss in a meeting, because we need to give a path to follow, if not every project have is own structure and when you open a project you don't know where to look to find objects.

IMHO Not easy, but must be done.

Reply:
Yes you are right. 
That's why I have corrected the file structure in my latest contribution and moved vendor to the root.
In your example, why should we have 2 directories? One for app and one for models? Keep it simple!

It's good to have a basic structure that makes a good fundament for small AND big projects.

Here's another approach: 
* config
    * db.php
    * init.php

* public_html
    * images
    * index.php
    * admin.php (if needed)

* projectfolder (could be named "app" for example)
    * Forms
        * ExampleForm1.php
        * ExampleForm2.php
        * UserDetailForm.php

    * Models
        * ExampleClass1.php
        * ExampleClass2.php
        * LoadAllUsers.php

    * Views
        * View1.php
        * View2.php
        * GridUserList.php

* vendor (contains all needed composer modules - don't touch them)
    * atk4
        * ...
    * autoload.php
    * ...

* composer.json